crates/rune-cli/src/main.rs

```diff
@@ -48,144 +48,103 @@
-            other => {
-                println!("Unrecognized option: {}", other);
-                args.help = true;
-            }
+    let args = {
+        let mut args = Args::from_args();
+        if args.dump {
+            args.dump_unit = true;
+            args.dump_stack = true;
+            args.dump_functions = true;
+            args.dump_types = true;
+            args.dump_native_functions = true;
+            args.dump_native_types = true;
         }
-    }
-
-    const USAGE: &str = "rune-cli [--trace] <file1> [, <file2> [, ..]] ";
-
-    if args.help {
-        println!("Usage: {}", USAGE);
-        println!();
-        println!("  --help, -h               - Show this help.");
-        println!(
-            "  --trace                  - Provide detailed tracing for each instruction executed."
-        );
-        println!("  --dump                   - Dump everything.");
-        println!("  --dump-unit              - Dump default information about unit.");
-        println!("  --dump-instructions      - Dump unit instructions.");
-        println!("  --dump-stack             - Dump the state of the stack after completion. If compiled with `--trace` will dump it after each instruction.");
-        println!("  --dump-functions         - Dump dynamic functions.");
-        println!("  --dump-types             - Dump dynamic types.");
-        println!("  --dump-native-functions  - Dump native functions.");
-        println!("  --dump-native-types      - Dump native types.");
-        println!("  --with-source            - Include source code references where appropriate (only available if -O debug-info=true).");
-        println!("  --experimental           - Enabled experimental features.");
-        println!();
-        println!("Compiler options:");
-        println!("  -O <option>       - Update the given compiler option.");
-        println!();
-        println!("Available <option> arguments:");
-        println!("  memoize-instance-fn[=<true/false>] - Inline the lookup of an instance function where appropriate.");
-        println!("  link-checks[=<true/false>]         - Perform linker checks which makes sure that called functions exist.");
-        println!("  debug-info[=<true/false>]          - Enable or disable debug info.");
-        println!("  macros[=<true/false>]              - Enable or disable macros (experimental).");
-        println!("  bytecode[=<true/false>]            - Enable or disable bytecode caching (experimental).");
-        return Ok(ExitCode::Success);
-    }
 
+        if args.dump_unit {
+            args.dump_unit = true;
+            args.dump_instructions = true;
+        }
+        if args.dump_functions
+            || args.dump_native_functions
+            || args.dump_stack
+            || args.dump_types
+            || args.dump_instructions
+        {
+            args.dump_unit = true;
+        }
+        args
+    };
+    let options = args.compiler_options.join(",").parse()?;
```

Comment:
Iterate over the options and parse them. Otherwise we're parsing commas again (see [this comment](https://github.com/rune-rs/rune/pull/51#issuecomment-691481403)).

This also means the [FromStr impl](https://github.com/rune-rs/rune/pull/51/files#diff-aca70ad2ef68c58ca9535c2da12b8ab3R20) is not needed.